src/stacked.rs

```diff
@@ -14,23 +16,51 @@ pub struct Stacked {
     fpu_regs: Option<StackedFpuRegs>,
 }
 
+fn bounds_check(bounds: Range<u32>, start: u32, len: u32) -> Result<(), ()> {
```

Comment:
`range.contains(&other_range)` (std API) didn't work so I wrote this

Reply:
Why didn't it work?

Reply:
trait bounds not met. `(0..4).contains(&2)` works but `(0..8).contains(&(2..4))` does not compile. I'm not aware of a std API that does "range contains range?"

Reply:
Ah, right